File: backend/app/schemas/document.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: UUID
    filename: str
    file_type: str
    file_size: int
    status: str
    created_at: datetime

    model_config = {
        "from_attributes": True
    }


class DocumentListResponse(BaseModel):
    documents: list[DocumentResponse]
    count: int
    max_sources: int = 5

File: frontend/src/types/document.ts
export interface Document {
  id: string;
  filename: string;
  file_type: string;
  file_size: number;
  status: string;
  created_at: string;
}

export interface DocumentListResponse {
  documents: Document[];
  count: number;
  max_sources: number;
}

export interface DocumentUploadResponse {
  message: string;
  document_id: string;
  filename: string;
  status: string;
}

export interface DocumentDeleteResponse {
  message: string;
}
